sqlmesh/core/audit/definition.py

```diff
@@ -340,6 +344,7 @@ def metadata_hash(self, audits: t.Dict[str, ModelAudit]) -> str:
             *sorted(self.tags),
             str(self.sorted_python_env),
             self.stamp,
+            self.fqn,
```

Comment:
why do audits have fqns? audits are not materialized so it shouldn't matter, it should probably just be the default catalog here. audits may not even need to define the fqn method

Reply:
That is different than how I was thinking about it. For example I have a `fqn` property on node even though node has no concept of default catalog nor makes assumptions about being a database object. The reason I did this is because the caller is basically saying "give me your fully qualified name" and for instances that is equal to name then they just return their name. This makes it so that you can call `fqn` on any node variant.